examples/step-69/step-69.cc

```diff
@@ -0,0 +1,2218 @@
+
+            const unsigned int n_face_q_points =
+              fe_face_values.get_quadrature().size();
+
+            for (unsigned int j = 0; j < dofs_per_cell; ++j)
+              {
+                if (!discretization->finite_element.has_support_on_face(j, f))
+                  continue;
+
+                /* Note that "normal" will only represent the contributions 
+                   from one of the faces in the support of the shape 
+                   function \phi_j. So we cannot normalize this local 
+                   contribution right here, we have to take it "as is" and pass 
+                   it to the copy data routine. */
+                Tensor<1, dim> normal;
+                if (id == Boundary::slip)
+                  {
+                    for (unsigned int q = 0; q < n_face_q_points; ++q)
+                      normal += fe_face_values.normal_vector(q) *
+                                fe_face_values.shape_value(j, q);
+                  }
+
+                const auto index = local_dof_indices[j];
+
+                Point<dim> position;
+                const auto global_index = partitioner->local_to_global(index);
+                for (unsigned int v = 0;
+                     v < GeometryInfo<dim>::vertices_per_cell;
+                     ++v)
+                  if (cell->vertex_dof_index(v, 0) == global_index)
+                    position = cell->vertex(v);
+
+                const auto old_id =
+                  std::get<1>(local_boundary_normal_map[index]);
+                local_boundary_normal_map[index] =
+                  std::make_tuple(normal, std::max(old_id, id), position);
+              } /* done with the loop on shape functions */
+          }     /* done with the loop on faces */
+      };        /* done with the definition of the worker */
+
+      /* This is the copy data routine for WorkStream */
+      const auto copy_local_to_global = [&](const auto &copy) {
+        const auto &is_artificial             = copy.is_artificial;
+        const auto &local_dof_indices         = copy.local_dof_indices;
+        const auto &local_boundary_normal_map = copy.local_boundary_normal_map;
+        const auto &cell_lumped_mass_matrix   = copy.cell_lumped_mass_matrix;
+        const auto &cell_cij_matrix           = copy.cell_cij_matrix;
+
+        if (is_artificial)
+          return;
+
+        for (const auto &it : local_boundary_normal_map)
+          {
+            auto &[normal, id, position] = boundary_normal_map[it.first];
+            auto &[new_normal, new_id, new_position] = it.second;
+
+            normal += new_normal;
+            id       = std::max(id, new_id);
+            position = new_position;
+          }
+
+        lumped_mass_matrix.add(local_dof_indices, cell_lumped_mass_matrix);
+
+        for (int k = 0; k < dim; ++k)
+          {
+            cij_matrix[k].add(local_dof_indices, cell_cij_matrix[k]);
+            nij_matrix[k].add(local_dof_indices, cell_cij_matrix[k]);
```

Comment:
is this a copy&paste error (using cell_cij twice)?

Reply:
No. I have nevertheless reorganized the code to simply copy the cij_matrix before renormalization.